<?xml version="1.0" ?>
<launch>

  <!-- gps receiver communication configuration -->
  <arg name="rtk_ip"              default="0.0.0.0"/>
  <arg name="rtk_port"            default="28001"/>

  <!-- Output frame_id -->
  <!-- This is the frame_id which corresponds to the output of the receiver and
       will therefore fill the output messages. If the transformations have not
       been setup internally on the reciever then this should be the location of
       the antenna (or main antenna if using a dual setup). If the
       transformation is done internally by the receiver, then this should be
       the location of the reference position setup in the receiver -->
  <arg name="output_frame_id"     default="base_link"/>


  <!-- Dual antenna settings -->
  <!-- Whether the gsof driver node should apply a heading offset due to the
       antennas not being in line with the robot. Leave False if this is done
       internally by the receiver or via a yaw offset in the frame_id.
       If true, the node will use the frame_id's below to offset the heading
       output -->
  <arg name="apply_dual_antenna_offset"   default="False"/>
  <!-- Antenna frame_ids to correct yaw direction if not done
       internally by the receiver. This is only used if we are applying the dual
       antenna offset. -->
  <arg name="gps_main_frame_id"   default="gps_main"/>
  <arg name="gps_aux_frame_id"    default="gps_aux"/>


  <node name="gsof_driver" pkg="trimble_gnss_driver" type="gsof_driver.py" output="screen" respawn="true">
    <param name="rtk_ip"              value="$(arg rtk_ip)"   type="string"/>
    <param name="rtk_port"            value="$(arg rtk_port)" type="int"/>
    <param name="gps_main_frame_id"   value="$(arg gps_main_frame_id)"/>
    <param name="gps_aux_frame_id"    value="$(arg gps_aux_frame_id)"/>
    <param name="output_frame_id"          value="$(arg output_frame_id)"/>
    <param name="apply_dual_antenna_offset" value="$(arg apply_dual_antenna_offset)"/>
    <!-- Output remapping -->
    <remap from="fix"                 to="gps/fix" /> <!-- fix output topic -->
    <remap from="yaw"                 to="yaw" /> <!-- yaw output topic if not using INS -->
    <remap from="attitude"            to="attitude" /> <!-- attitude output topic, includes yaw if using INS -->
  </node>


</launch>
